src/findboundaries.jl

```diff
@@ -0,0 +1,22 @@
+"""
+```
+bordered_img = findboundries(img)
+bordered_img = findboundries(img, background)
+```
+Returns a copy of the original image where boundries between labeled regions are true.
+Parameters:
+ -  img             = Binary/Grayscale input image
+ -  background      = Value to be given to pixels/elements that are cleared (Default value is 0)
+"""
+
+function findboundaries(img::AbstractArray, background::Integer=0)
+
+    connectivity = ntuple(i -> 3, ndims(img))
+    labels = label_components(img,trues(connectivity))
+
+    boundaries = dilate(img) .!= erode(img)
+    foreground_image = (labels .!= background)
+    boundaries = boundaries .& foreground_image
```

Comment:
While this makes for very elegant and compact code, I believe there is a more efficient approach. 

A pixel is on the boundary of a connected component if at least one 8-connected neighbour belongs to the background, and one does not. 

One can therefore loop over the image once, while checking the 8-neighbourhood of each pixel for the necessary condition, and populate the boundaries array on the go.

Reply:
Unsure about this, my best guess is that memory allocation for `Array{Bool}` won't be a hot spot. Just a guess.

Reply:
Presently, the code is looping over the whole image to perform a dilation, and then once again over the whole image to perform an erosion. Then it is looping over the whole image again to determine what the foreground is, and finally once again to determine the boundaries. 

I am suggestion that three of those loops over the entire image are unnecessary if one doesn't try to write the boundary detection in terms of erosion and dilation operations.

Reply:
One fast approach is to use a convolution. To do this with integers:
`edge_kern = [1 1 1;
                      1 10 1;
                      1 1 1]
surface = conv(im, edge_kern)[2:end-1, 2:end-1]`
The resulting image highlights surface pixels as having a range of 10:17. Logically then one could reimplement a convolution where we drop the value directly into a waiting array at the centre of the window.
More lazily you can use findall:
`wb = findall(in(10:17), surface)`
Which is good if you want sparse data.